core/src/ecs/query/filter.cpp

```diff
@@ -134,6 +134,9 @@ void QueryFilter::update()
 
     // Update the pin masks with which each node will run.
     mNodePins[0] = 0;
+    // Copy mNodeCount since the compiler thinks it will change.
+    int mNodeCount = this->mNodeCount;
+    CUBOS_ASSERT(mNodeCount <= QueryNode::MaxCursorCount);
     for (int i = 1; i < mNodeCount; ++i)
```

Comment:
```suggestion
    // Copy mNodeCount since the compiler thinks it will change. See issue TODO add number here
    int nodeCount = this->mNodeCount;
    CUBOS_ASSERT(nodeCount <= QueryNode::MaxCursorCount);
    for (int i = 1; i < nodeCount; ++i)
```
I would just change the naming of the variable (since we reserve the m prefix for member variables), and maybe link the issue you created in the comment above.